api/envoy/extensions/http/header_validators/envoy_default/v3/header_validator.proto

```diff
@@ -115,4 +135,8 @@ message HeaderValidatorConfig {
   // Envoy will respond with 400 to requests with disallowed methods.
   // By default methods with arbitrary names are accepted.
   bool restrict_http_methods = 3;
+
+  // Action to take when a client request with a header name containing underscore characters is received.
+  // If this setting is not specified, the value defaults to ALLOW.
+  HeadersWithUnderscoresAction headers_with_underscores_action = 4;
```

Comment:
Does this field need an Envoy build with `uhv=enabled` to take effect? If so please document.

Reply:
UHV is enabled in [HCM](https://github.com/envoyproxy/envoy/blob/c81f38a0ee967f22b16fea040ea61b42d8034140/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L804)

which is currently marked as hidden.

The `uhv=enabled` is really a temporary build flag to safely transition to the new verification code. Do you think we need to document it? It will not be added into the generated documentation anyway because of the hidden flag.